cocos/2d/CCNode.cpp

```diff
@@ -950,6 +958,8 @@ void Node::addChildHelper(Node* child, int localZOrder, int tag, const std::stri
     child->setParent(this);
     child->setCameraMask(getCameraMask());
 
+    child->_setOrderOfArrival(s_globalOrderOfArrival++);
```

Comment:
why not do `s_globalOrderOfArrival++` internally in the implementation of `Node::_setOrderOfArrival()`?